pkg/infrastructure/gcp/clusterapi/firewallrules.go

```diff
@@ -198,6 +225,27 @@ func createFirewallRules(ctx context.Context, in clusterapi.InfraReadyInput, net
 		return err
 	}
 
+	// etcd are needed for master communication for etcd nodes
+	firewallName = fmt.Sprintf("%s-etcd", in.InfraID)
+	srcTags = []string{masterTag}
+	targetTags = []string{masterTag}
+	srcRanges = []string{}
+	if err := addFirewallRule(ctx, firewallName, network, projectID, getEtcdPorts(), srcTags, targetTags, srcRanges); err != nil {
```

Comment:
For a follow-up, I'd recommend defining a `struct` for the input needed by this function:
```go
type FirewallRuleInput struct {
    firewallName string
    networkName string
    projectID string
    allowedPorts []*compute.FirewallAllowed
    sourceTags []string
    targetTags []string
    sourceRanges []string
}
```
Then the usage becomes:
```go
if err := addFirewallRule(ctx, &firewallRuleInput{
    firewallName: firewallName,
    networkName: network,
    projectID: projectID,
    allowedPorts: getEtcdPorts(),
    sourceTags: srcTags,
    targetTags: targetTags,
    sourceRanges: srcRanges,
}); err != nil {
```
This also has the benefit that you can add more optional fields to the struct in the future without having the change every place where `addFirewallRule` is called.